packer/build.pkr.hcl

```diff
@@ -10,6 +10,7 @@ build {
       "--scp-extra-args", "'-O'",
       "-e", "aws_region=${var.aws_region}",
       "-e", "artifactory_version=${var.artifactory_version}",
+      "-e", "artifactory_var_device_node=/dev/nvme1n1",
```

Comment:
Consider using the [ansible-role-aws-nvme-device-files](https://github.com/companieshouse/ansible-role-aws-nvme-device-files) role to guarantee stable device nodes. The `nvme*` device nodes will still be used internally (and exposed by tools such as `lsblk`) but `udev` rules introduced by the role will ensure that symbolic links are created with names matching those specified in the block devices used at instance launch time (i.e. those specified in `launch_block_device_mappings` blocks).

This means you could introduce a variable for the device path:

```hcl
variable "data_volume_device_node" {
  type        = string
  default     = "/dev/sdb"
  description = "The device node path for the data volume"
}
```

Which could be used in block device mappings in Packer:

```hcl
 launch_block_device_mappings {
    device_name = var.data_volume_device_node
    ...
  }

```

And the same variable can be fed to Ansible:

```
"-e", "artifactory_var_device_node=${var.data_volume_device_node}",
```

Note the `ansible-role-aws-nvme-device-files` role does require Python 2.7/Python 3 on the remote but is based on a set of udev rules and Amazon-provided scripts that use device-specific metadata to guarantee stable device nodes.

Reply:
Updated in latest commits